goth/runner/probe/__init__.py

```diff
@@ -351,7 +352,13 @@ async def run_command_on_host(
         command: str,
         env: Optional[Mapping[str, str]] = None,
         command_timeout: float = 300,
-    ) -> AsyncIterator[Tuple[asyncio.Task, PatternMatchingEventMonitor]]:
+        get_process_monitor: bool = False,
+    ) -> AsyncIterator[
+        Union[
+            Tuple[asyncio.Task, PatternMatchingEventMonitor],
+            Tuple[asyncio.Task, PatternMatchingEventMonitor, process.ProcessMonitor],
+        ]
+    ]:
```

Comment:
I think it's safe to always include the `ProcessMonitor` in the output and omit the `get_process_monitor` argument. This way the interface stays unchanged and the return type gets cleaner.

Reply:
Wouldn't that hurt all the current tests that use `run_command_on_host`?

Reply:
You're right - these would need to be updated to reflect this. It's not a big problem though - we can consider this a breaking change and make a minor release of `goth` and update the usage in `yapapi`.

I think it's better than introducing another optional argument to `run_command_on_host` and making its return type complex.

Using an analogy - we could have *yet another* optional argument like `get_asyncio_task` and have the result tuple include the `Task` instance or not, depending on that argument's value.

Reply:
I like this reasoning!